### PR TITLE
fixed: K8sInstallerConfig CR are not getting deleted when byomachine is deleted

### LINKED ISSUE #547: Byoh hostinfo.architecture is not matching with the supported bundle name
Byoh `hostinfo.architecture`  is not matching with the supported bundle name.   

byoh agent is setting architecture using `runtime.GOARCH` which provide `amd64` but the bundle name support `x86-64` means, we need to convert `amd64` to `x86-64` in InstallerConfig controller when generating the installation script.

Part of issue: #467

### LINKED ISSUE #548: K8sInstallerConfig CR are not getting deleted when byomachine is deleted
**What steps did you take and what happened:**
First create a cluster with byomachine infraRef as K8sInstallerConfigTemplate.
Next try to delete the cluster, the k8sinstallerconfig CRs created byomachine controller won't be deleted as they are trying to lookup for the byomachine which owns them but getting byomachine not found error.
byomachine is not waiting for installer CR to be deleted first before removing its finalizer.

**What did you expect to happen:**
On deleting the byomachine, the k8sinstallerconfig ,owns by respective byomachine, should be delete as well.

Belongs to #467

### PR DESCRIPTION
**Pull request by mayur-tolexo**
**What this PR does / why we need it**:
If installer resource owner byomachine not found and delete timestamp is set then also doing the delete reconciliation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #548

### CODE REVIEW

**Review by shubham14bajpai — APPROVED** [review]
LGTM. I suggestion: Can we remove the deletion of secret from `reconcileDelete`? Reason being that secret does not have any finalizers and the ownerRef of installerConfig so it should get the delete call automatically.

**Review by shubham14bajpai — APPROVED** [review]

File: controllers/infrastructure/k8sinstallerconfig_controller.go [hunk 1]
```diff
@@ -258,28 +258,6 @@ func (r *K8sInstallerConfigReconciler) ByoMachineToK8sInstallerConfigMapFunc(o c
 func (r *K8sInstallerConfigReconciler) reconcileDelete(ctx context.Context, scope *k8sInstallerConfigScope) (reconcile.Result, error) {
 	logger := scope.Logger
 	logger.Info("Deleting K8sInstallerConfig")
-
```
Comment (dharmjit) [inline controllers/infrastructure/k8sinstallerconfig_controller.go:261]: Are we relying on the K8s garbage collection mechanism to clean this?

**Review by dharmjit — APPROVED** [review]
LGTM

File: controllers/infrastructure/k8sinstallerconfig_controller.go (hunk 1 above)
Comment (mayur-tolexo) [reply controllers/infrastructure/k8sinstallerconfig_controller.go:261]: Yup, we have owner reference set in the secret created by installer.

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**